frontend/src/components/TopBar.js

```diff
@@ -1,12 +1,23 @@
 import React from 'react';
 import { useNavigate } from 'react-router-dom';
+import { useAuth } from '../contexts/AuthContext';
 import BusyBeaverNoBG from '../assets/BusyBeaverNoBG.png';
 import { Button, Typography } from '@mui/material';
 
 const TopBar = () => {
   const navigate = useNavigate();
+  const { logout } = useAuth(); // Destructure logout from useAuth
   const handleNavigateToHome = () => navigate('/home');
-  const handleLogout = () => navigate('/');
+  
+  const handleLogout = async () => {
+    try {
+      await logout();
+      navigate('/'); // Redirect to splash page
+    } catch (error) {
+      console.error("Failed to log out", error);
```

Comment:
Could we handle this with some user feedback instead of the console?

Reply:
What would that look like?

Reply:
Error message next to the logout button possibly, or maybe a modal dialog? I'm assuming it would be rare but in this case the user could be spamming the button and not seeing anything happen.

I'm putting up a modal dialog now for deleting a project if you want an example of that.